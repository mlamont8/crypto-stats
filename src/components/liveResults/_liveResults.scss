.liveItem {
  font-size: 18px;
}

#arrow-up {
  color: green;
}
#arrow-down {
  color: red;
}
.conversion {
  font-size: 16px;
}

// .current h1 {
//   font-size: 20px;
// }

// .current h2 {
//   font-size: 14px;
//   margin: 4px;
// }

.toPrice {
  padding-left: 2px;
  font-size: 8px;
}
