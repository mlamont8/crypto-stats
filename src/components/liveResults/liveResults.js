import React from "react";
import FontAwesomeIcon from "@fortawesome/react-fontawesome";
import PropTypes from "prop-types";

const LiveResults = props => {
  const { flag, dollar, exchange, price, to } = props;
  const conversion = (price * dollar).toFixed(2);
  const arrow = flag === "2" ? "arrow-down" : "arrow-up";
  return (
    <div className="current text-center">
      <h3>CURRENT</h3>
      <FontAwesomeIcon icon={["fas", "coins"]} size="2x" />
      {exchange}
      <hr />
      <div className="row">
        <div className="liveItem">
          <span className="price">{price}</span>
          <span className="toPrice">{to}</span>
        </div>
      </div>
      <div className="row">
        <span className="conversion" id={arrow}>
          (${conversion})
        </span>
        <FontAwesomeIcon icon={["fas", arrow]} />
      </div>
    </div>
  );
};

LiveResults.propTypes = {
  price: PropTypes.string,
  flag: PropTypes.string,
  exchange: PropTypes.string,
  dollar: PropTypes.number,
  to: PropTypes.string
};

LiveResults.defaultProps = {
  price: "",
  flag: "",
  exchange: "",
  dollar: 0,
  to: ""
};

export default LiveResults;
