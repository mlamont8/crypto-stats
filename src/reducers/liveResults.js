import { liveTime } from "../helpers";

const initialState = [
  {
    id: 0,
    searchesThisSession: 0,
    time: ""
  }
];
const liveResults = (state = initialState, action) => {
  switch (action.type) {
    case "INCOMING_LIVE_UPDATE":
      return [
        ...state,
        {
          searchesThisSession: action.searchesThisSession,
          id: action.id,
          flag: action.update.flag,
          price: action.update.price,
          time: liveTime(action.update.time)
        }
      ];

    default:
      return state;
  }
};

export default liveResults;
