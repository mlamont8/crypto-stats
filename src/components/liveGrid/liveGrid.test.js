import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import liveResults from "./liveGrid";

describe("LiveGrid", () => {
  const props = {
    liveResults: [
      {
        searchesThisSession: 1,
        id: 1,
        flag: "0",
        price: "0.002498",
        time: "8:57:59 PM"
      },
      {
        searchesThisSession: 1,
        id: 2,
        flag: "1",
        price: "0.002498",
        time: "8:58:59 PM"
      },
      {
        searchesThisSession: 1,
        id: 3,
        flag: "1",
        price: "0.002498",
        time: "8:59:59 PM"
      },
      {
        searchesThisSession: 2,
        id: 1,
        flag: "1",
        price: "0.002498",
        time: "9:00:59 PM"
      }
    ],
    usd: 7702.12,
    to: "BTC"
  };
  it("renders without crashing", () => {
    shallow(<liveResults {...props} />);
  });

  it("matches snapshot", () => {
    const wrapper = renderer.create(<liveResults {...props} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
