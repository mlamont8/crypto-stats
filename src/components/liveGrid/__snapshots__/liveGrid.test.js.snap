// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiveGrid matches snapshot 1`] = `
<liveResults
  liveResults={
    Array [
      Object {
        "flag": "0",
        "id": 1,
        "price": "0.002498",
        "searchesThisSession": 1,
        "time": "8:57:59 PM",
      },
      Object {
        "flag": "1",
        "id": 2,
        "price": "0.002498",
        "searchesThisSession": 1,
        "time": "8:58:59 PM",
      },
      Object {
        "flag": "1",
        "id": 3,
        "price": "0.002498",
        "searchesThisSession": 1,
        "time": "8:59:59 PM",
      },
      Object {
        "flag": "1",
        "id": 1,
        "price": "0.002498",
        "searchesThisSession": 2,
        "time": "9:00:59 PM",
      },
    ]
  }
  to="BTC"
  usd={7702.12}
/>
`;
